run_dir/static/js/pricing_quote.js

```diff
@@ -68,12 +76,60 @@ app.component('v-pricing-quote', {
         reset_special_addition() {
             this.$root.quote_special_addition_label = ''
             this.$root.quote_special_addition_value = 0
+        },
+        update_freetext: function(e) {
+          this.message = e.target.value;
+          },
+        generate_quote:  function (event) {
+          product_list = {}
+          if(Object.keys(this.$root.quote_prod_ids).length > 0){
+            for (prod_id in this.$root.quote_prod_ids){
+              product_list[prod_id] = this.$root.all_products[prod_id]
+              product_list[prod_id]['product_cost'] = this.$root.productCost(prod_id)[this.$root.price_type].toFixed(2)
+            }
+            product_list['total_products_cost'] = this.product_cost_sum[this.$root.price_type].toFixed(2)
+            product_list['total_cost'] = this.quote_cost[this.$root.price_type]
+            product_list['price_type'] = this.$root.price_type
+            if (this.any_special_addition){
+              var obj = {};
+              obj[this.$root.quote_special_addition_label] = this.$root.quote_special_addition_value;
+              product_list['special_addition'] =  obj
+            }
+            if (this.any_special_percentage){
+              var obj = {};
+              obj[this.$root.quote_special_percentage_label] = this.$root.this.$root.quote_special_percentage_value
+              product_list['special_percentage'] = obj
+            }
+            if(this.message !== ''){
+              product_list['agreement_summary'] = this.message
+            }
+
+            var newForm = $('<form>', {
```

Comment:
Maybe add a little comment why this is needed

Reply:
Updated!